src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -83,21 +97,42 @@ public ServiceBusJmsConnectionFactory(ConnectionStringBuilder connectionStringBu
      */
     public ServiceBusJmsConnectionFactory(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
         this.settings = settings;
+        this.sasKey = sasKey;
+        this.sasKeyName = sasKeyName;
         this.initialize(sasKeyName, sasKey, host, settings);
     }
     
-    private void initialize(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
-        if (sasKeyName == null || sasKeyName == null || host == null) {
-            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
+    /**
+     * Create a ServiceBusJmsConnectionFactory using a credential and host name.
+     * @param Credential. A token provider credential that will be use to acquire an aad token. For default credential ether pass null or DefaultAzure
```

Comment:
>DefaultAzure [](http://example.com/codeflow?start=138&length=12)

DefaultAzureCredential

Reply:
Removed since we are no longer implementing default reader just accepting it as a valid credential type to get token